### PR TITLE
Measure #24

### PR DESCRIPTION
**Pull request by PatHock**
Fleshed out Note model and definitions file. 
This should have been in the note branch!

### CODE REVIEW

File: src/main/java/com/micetr0/definitions/Defs.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+            this.frequency = frequency;
+        }
+
+        public Double getFrequency() {
+            return this.frequency;
+        }
+
+    }
+    public enum Key {
+        // Uses 7 notes per octave scales
```
Comment (rbosseIII) [inline src/main/java/com/micetr0/definitions/Defs.java]: document how implemented

File: src/main/java/com/micetr0/model/Measure.java [hunk 2]
```diff
@@ -6,9 +6,39 @@
  */
 
 package com.micetr0.model;
+import java.util.ArrayList;
+import java.util.LinkedList;
```
Comment (rbosseIII) [inline src/main/java/com/micetr0/model/Measure.java]: remove

File: src/main/java/com/micetr0/model/Measure.java [hunk 3]
```diff
@@ -6,9 +6,39 @@
...
+import java.lang.Exception;
+import java.util.List;
 
 public class Measure {
+    private Double sizeLimit;
+    private List<Note> notes = new ArrayList<Note>();
```
Comment (rbosseIII) [inline src/main/java/com/micetr0/model/Measure.java]: remove

File: src/main/java/com/micetr0/model/Measure.java [hunk 4]
```diff
@@ -6,9 +6,39 @@
...
+    private int currIndex;
 
-    //TODO: Implement
+
+    public Double getSizeLimit() {
+        return sizeLimit;
+    }
+
+    public List<Note> getNotes() {
```
Comment (rbosseIII) [inline src/main/java/com/micetr0/model/Measure.java]: remove

File: src/main/java/com/micetr0/model/Measure.java [hunk 5]
```diff
@@ -6,9 +6,39 @@
...
+        return notes;
+    }
+
+    public void setSizeLimit(Double sizeLimit) {
+        this.sizeLimit = sizeLimit;
+    }
+
+    public void setNotes(List<Note> notes) {
```
Comment (rbosseIII) [inline src/main/java/com/micetr0/model/Measure.java]: dumb, get rid of. blame rick

File: src/main/java/com/micetr0/model/Measure.java [hunk 6]
```diff
@@ -6,9 +6,39 @@
...
+        this.notes = notes;
+    }
+
+    public int getCurrIndex() {
```
Comment (rbosseIII) [inline src/main/java/com/micetr0/model/Measure.java]: remove

File: src/main/java/com/micetr0/model/Note.java [hunk 7]
```diff
@@ -4,11 +4,130 @@
...
+     * Every Note object should have all of these attributes to be stored in the database
+     * @param type          The type of the note (ex: Sixteenth)
+     * @param pitch         The specific Note, with reference A0 = 440 Hz
+     * @param measureIndex  The position of a note within a measure, 0 being the first note in the measure
+     * @param measureId     The unique ID of the measure that contains this note
+     * @param sectionId     The unique ID of the section that contains this note
+     * @param compositionId The unique ID of the composition that contains this note
+     */
+    public Note(Defs.NoteType type, Defs.Pitch pitch, Integer measureIndex, Integer measureId, Integer sectionId, Integer compositionId) {
+    //empty constructor
```
Comment (rbosseIII) [inline src/main/java/com/micetr0/model/Note.java]: not empty

File: src/test/java/com/micetr0/controller/MeasureControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Before
+    public void setUp()
+    {
+        controller = new MeasureController();
+        model = new Measure();
+        controller.setUp(model);
+        controller.updateMaxLen(1);
+    }
+
```
Comment (rbosseIII) [inline src/test/java/com/micetr0/controller/MeasureControllerTest.java]: need test cases

File: src/test/java/com/micetr0/controller/NoteControllerTest.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package com.micetr0.controller;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class NoteControllerTest {
+
+    @Test
+    public void addNote() {
```
Comment (rbosseIII) [inline src/test/java/com/micetr0/controller/NoteControllerTest.java]: need test cases

File: src/test/java/com/micetr0/model/MeasureTest.java [hunk 10]
```diff
@@ -0,0 +1,11 @@
+package com.micetr0.model;
+
+import org.junit.Test;
+import com.micetr0.model.Note;
+
+import static org.junit.Assert.*;
+
+public class MeasureTest {
+//    private Note note = new Note();
```
Comment (rbosseIII) [inline src/test/java/com/micetr0/model/MeasureTest.java]: need test cases

File: src/main/java/com/micetr0/model/Measure.java (hunk 2 above)
Comment (PatHock) [reply src/main/java/com/micetr0/model/Measure.java]: Removed unused imports

File: src/main/java/com/micetr0/definitions/Defs.java (hunk 1 above)
Comment (PatHock) [reply src/main/java/com/micetr0/definitions/Defs.java]: Constructor, getPitches methods have JavaDoc comments

File: src/main/java/com/micetr0/model/Measure.java (hunk 5 above)
Comment (PatHock) [reply src/main/java/com/micetr0/model/Measure.java]: Deleted everything in measure class. Starting over

File: src/main/java/com/micetr0/model/Measure.java (hunk 4 above)
Comment (PatHock) [reply src/main/java/com/micetr0/model/Measure.java]: removed

File: src/main/java/com/micetr0/model/Measure.java (hunk 3 above)
Comment (PatHock) [reply src/main/java/com/micetr0/model/Measure.java]: removed

File: src/main/java/com/micetr0/model/Measure.java (hunk 6 above)
Comment (PatHock) [reply src/main/java/com/micetr0/model/Measure.java]: removed

File: src/main/java/com/micetr0/model/Note.java (hunk 7 above)
Comment (PatHock) [reply src/main/java/com/micetr0/model/Note.java]: updated

File: src/test/java/com/micetr0/controller/MeasureControllerTest.java (hunk 8 above)
Comment (PatHock) [reply src/test/java/com/micetr0/controller/MeasureControllerTest.java]: Measure controller hasn't been implemented yet, will be taken care of in future pull request

File: src/test/java/com/micetr0/controller/NoteControllerTest.java (hunk 9 above)
Comment (PatHock) [reply src/test/java/com/micetr0/controller/NoteControllerTest.java]: Note controller still has to be implemented. Will be addressed in future pull request

File: src/test/java/com/micetr0/model/MeasureTest.java (hunk 10 above)
Comment (PatHock) [reply src/test/java/com/micetr0/model/MeasureTest.java]: Measure still not implemented. Will be addressed in future pull request.

**Pull request merged**